proto/src/depscloud/api/v1beta/core.proto

```diff
@@ -0,0 +1,53 @@
+//
+// This file attempts to follow common styles and design patterns described by
+// Google Cloud based on their extensive use of protocol buffers.
+//
+// - See here for more: https://cloud.google.com/apis/design/design_patterns
+//
+syntax = "proto3";
+
+package depscloud.api.v1beta;
+
+option go_package = "github.com/depscloud/api/v1beta";
+
+// A source in depscloud represents an origin for information. This includes,
+// but is not limited to, repositories and artifactories.
+message Source {
+    // The address of the source, typically a git url or artifact path.
+    string url = 1;
+    // A ref reference for this source, typically a branch name or version.
+    string ref = 2;
+    // The kind of source, typically repository or artifactory.
+    string kind = 3;
+}
+
+// A module in depscloud is a generic representation of an application or library.
+message Module {
```

Comment:
I wonder if it would be useful to add some sort of key/value map to track things like module maintainers, ownership, license, etc. -- the kind of metadata that could help answer questions like "who do I talk to about this?" or "could I switch from GPL to Apache?" more quickly.

Reply:
I really liked the notion of "labels" (seen in the design patterns guide). I'll add them to all data types.

Reply:
I've added the concept of labels to each element